models/user/userModel.js

```diff
@@ -0,0 +1,66 @@
+import mongoose, { Schema } from 'mongoose';
+import { addressSchema, orderItemSchema } from '../schemas';
+import { accountStatuses, roles } from '../../constants';
+import { phoneNumberPattern } from '../../helpers';
+
+
+const ObjectId = Schema.Types.ObjectId;
+
+const userSchema = new Schema(
+  {
+    fullName: {
+      type: String,
+      required: [true, 'Full name is required'],
+      minlength: [3, 'Full name must be at least 3 characters long'],
+    },
+    password: {
+      type: String,
```

Comment:
maybe add pattern validation for obligatory special characters, uppercase and lowercase letters, etc

Reply:
Okay, it will be done

Reply:
Added